### ISSUE TITLE
GetAsync_ResponseContentAfterClientAndHandlerDispose_Success failed in CI on Linux

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#3749]
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/3215/console
```
10:54:28    System.Net.Http.Functional.Tests.HttpClientHandlerTest.GetAsync_ResponseContentAfterClientAndHandlerDispose_Success [FAIL]
10:54:28       Assert.True() Failure
10:54:28       Expected: True
10:54:28       Actual:   False
10:54:28       Stack Trace:
10:54:28 Finished:    System.Net.Http.Unit.Tests
10:54:28 
10:54:28 === TEST EXECUTION SUMMARY ===
10:54:28    System.Net.Http.Unit.Tests  Total: 783, Errors: 0, Failed: 0, Skipped: 0, Time: 2.503s
10:54:28             at System.Net.Http.Functional.Tests.HttpClientHandlerTest.<AssertSuccessfulGetResponse>d__0.MoveNext()
10:54:28          --- End of stack trace from previous location where exception was thrown ---
10:54:28             at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
10:54:28             at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
10:54:28             at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
10:54:28             at System.Net.Http.Functional.Tests.HttpClientHandlerTest.<GetAsync_ResponseContentAfterClientAndHandlerDispose_Success>d__14.MoveNext()
10:54:28          --- End of stack trace from previous location where exception was thrown ---
10:54:28             at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
10:54:28             at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
10:54:28          --- End of stack trace from previous location where exception was thrown ---
10:54:28             at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
10:54:28             at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
10:54:28          --- End of stack trace from previous location where exception was thrown ---
10:54:28             at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
10:54:28             at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
10:54:44 Finished:    System.Net.Http.Functional.Tests
10:54:44 
10:54:44 === TEST EXECUTION SUMMARY ===
10:54:44    System.Net.Http.Functional.Tests  Total: 254, Errors: 0, Failed: 1, Skipped: 0, Time: 19.002s
```

**Comment by vijaykota** [comment]
The dev code looks fine. I also tried reproing by tweaking the test case to use HttpCompletionOption.ResponseHeadersRead so that the first Read on HttpResponse happens after HttpClient and HttpClientHandler have been Dispose'd. But the test case passes since the code is in order.

The code where the failure was reported has been removed and used to be:
````
                Assert.Equal<HttpStatusCode>(HttpStatusCode.OK, response.StatusCode);
                Assert.Equal<string>("OK", response.ReasonPhrase);
                string responseContent = await response.Content.ReadAsStringAsync();
                Assert.True(TestHelper.JsonMessageContainsKeyValue(responseContent, "url", uri.AbsoluteUri));
                output.WriteLine(responseContent);
````
The failing Assert corresponds to responseContent not containging "url.." but since the output.WriteLine is after the Assert, the content was not outputted. This has since been rectified in TestHelper.VerifyResponseBody by outputting the responseContent before the Assert. So if this ever reproes, we will have more information.
